sample-ios-app/sample-ios-app/ViewController.swift

```diff
@@ -31,7 +31,7 @@ class ViewController: UIViewController {
     }
     
     private class KittenView: AbstractKittenView {
-        private let viewController: ViewController
+        private weak var viewController: ViewController?
```

Comment:
So the `KittenBinder` accepts the `KittenView` and holds hard reference to it unless `onViewDestroyed` is called.

First thing we could fix is to not hold the `KittenView` in the `ViewController`:
`kittenBinder.onViewCreated(view: KittenView(viewController: self))`

After that there still will be a retain loop: `ViewController` -> `KittenBinder` -> `KittenView` -> `ViewController`. We can fixed this by destroying the `KittenBinder` not in `deinit` but in `viewWillUnload()`. The `KittenBinder` will clear the reference to the `KittenView`.

Reply:
viewWillUnload mark as [deprecated](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621427-viewwillunload?language=objc)

Reply:
I told with ios dev, and he said that we should use weak for reference to VC.

Reply:
In our case the hard reference to view is cleared when the view is destroyed. What do you think about the following approach?
```swift
    private let kittenBinder = KittenBinder(storeBuilder: KittenStoreBuilderImpl())
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        kittenBinder.onViewCreated(view: KittenView(viewController: self))
        kittenBinder.onStart()
    }
        
    override func viewDidDisappear(_ animated: Bool) {
        kittenBinder.onStop()
        kittenBinder.onViewDestroyed()
        super.viewDidDisappear(animated)
    }
    
    deinit {
        kittenBinder.onDestroy()
    }
```

Reply:
This solution is working, but should we recreate KittenView so often?

Reply:
Is there any noticeable performance or memory impact? The KittenViee class itself is very lightweight, it is just a wrapper around the ViewConttoller.

Reply:
I don't think that we have a problem with memory, but what is the reason why we need to recreate it? I can push changes with your approach today a nit later.

Reply:
Well. Because we don't need to draw the view when it is disappeared. So we can actually release some memory when the ViewConttoller is invisible. And it also avoids memory leaks without `weak` and question marks.

Reply:
Like in Android fragments, onDestroyView is called when the fragment is pushed to the back stack

Reply:
yes but viewDidDisappear is the same like onPause or onStrop not onDestroyView and I think it is not the same :)

Reply:
From Apple docs:

> Notifies the view controller that its view was removed from a view hierarchy.

So the view is removed from the hierarchy, is like pushing the fragment to the back stack.

Anyway, the KittenView is just a wrapper. The actual view is in the ViewController and is not recreated.